ideascube/widgets.py

```diff
@@ -1,6 +1,73 @@
 from django.forms import widgets
 
 from ideascube.configuration import get_config
+from django.utils.html import mark_safe, escape
+from django.utils.encoding import force_text
+from django.forms.utils import flatatt
+from django.utils.translation import get_language
+
+import bleach
+
+
+_ideascube_to_tinyMCE_language_map = {
+    'fr': 'fr_FR'
+}
+
+
+class RichTextEntry(widgets.Widget):
+    def __init__(self, *,
+            with_media = False,
+            attrs=None):
+        super().__init__(attrs=attrs)
+        self.with_media = with_media
+
+        self.authorized_tags = ['p', 'a', 'ul', 'ol', 'li', 'br',
+                                'h2', 'h3', 'h4', 'h5']
+        self.authorized_attributes = {
+            'a': ['href', 'title'],
+            'img': ['src', 'width', 'height', 'alt'],
+            'iframe': ['src', 'width', 'height', 'allowfullscreen']
+        }
+        if with_media:
+            self.authorized_tags += ['img', 'iframe']
+
+    def get_language(self):
+        ideascube_language = get_language()
+        return _ideascube_to_tinyMCE_language_map.get(
+            ideascube_language, ideascube_language)
+
+    def render(self, name, value, attrs=None):
+        if value is None:
+            value = ''
+
+        # The widget displayed to the user
+        div_attrs = {'id': name,
+                     'class': 'tinymce_editor',
+                     'tinymce_use_media': self.with_media,
+                     'tinymce_language_code': self.get_language()
```

Comment:
AFAIK adding custom attributes has never been valid HTML (or maybe it's XHTML?), unless you were using the "transitional" DTD and maybe other subtleties.

HTML5 added the `data-*` attributes specifically for this purpose.